shardingsphere-proxy/shardingsphere-proxy-backend/src/test/java/org/apache/shardingsphere/proxy/backend/handler/distsql/ral/queryable/ConvertYamlConfigurationHandlerTest.java

```diff
@@ -0,0 +1,136 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.shardingsphere.proxy.backend.handler.distsql.ral.queryable;
+
+import lombok.SneakyThrows;
+import org.apache.shardingsphere.distsql.parser.statement.ral.queryable.ConvertYamlConfigurationStatement;
+import org.apache.shardingsphere.mode.manager.ContextManager;
+import org.apache.shardingsphere.proxy.backend.context.ProxyContext;
+import org.apache.shardingsphere.proxy.backend.response.header.query.QueryHeader;
+import org.apache.shardingsphere.proxy.backend.response.header.query.QueryResponseHeader;
+import org.apache.shardingsphere.proxy.backend.session.ConnectionSession;
+import org.apache.shardingsphere.proxy.backend.util.ProxyContextRestorer;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.IOException;
+import java.sql.SQLException;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.assertThat;
+import static org.junit.Assert.assertFalse;
+import static org.mockito.Mockito.RETURNS_DEEP_STUBS;
+import static org.mockito.Mockito.mock;
+
+public class ConvertYamlConfigurationHandlerTest extends ProxyContextRestorer {
+    
+    private final String resourceFilePath = "/conf/convert/config-resource.yaml";
+    
+    private final String shardingFilePath = "/conf/convert/config-sharding.yaml";
+    
+    private final String resourceExpectedFilePath = "/expected/convert-add-resource.yaml";
+    
+    private final String shardingExpectedFilePath = "/expected/convert-create-sharding.yaml";
+    
+    private final String resource = "resource";
+    
+    private final String sharding = "sharding";
+    
+    private final Map<String, String> featureMap = new HashMap<>(2, 1);
+    
+    @Before
+    public void setup() {
+        featureMap.put(resource, resourceFilePath);
+        featureMap.put(sharding, shardingFilePath);
+    }
+    
+    @Before
+    public void init() {
+        ContextManager contextManager = mock(ContextManager.class, RETURNS_DEEP_STUBS);
+        ProxyContext.init(contextManager);
+    }
+    
+    @Test
+    public void assertExecuteWithAddResource() throws SQLException {
+        assertExecute(resource, resourceExpectedFilePath);
+    }
+    
+    @Test
+    public void assertExecuteWithCreateSharding() throws SQLException {
+        assertExecute(sharding, shardingExpectedFilePath);
+    }
+    
+    public void assertExecute(final String feature, final String expectedFilePath) throws SQLException {
```

Comment:
`feature` is not a good name for `resource`, maybe change to `type`?

Reply:
fixed~